artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/connect/mirror/AMQPMirrorControllerAggregation.java

```diff
@@ -0,0 +1,118 @@
+import java.util.List;
+
+import org.apache.activemq.artemis.api.core.Message;
+import org.apache.activemq.artemis.api.core.QueueConfiguration;
+import org.apache.activemq.artemis.api.core.SimpleString;
+import org.apache.activemq.artemis.core.server.ActiveMQComponent;
+import org.apache.activemq.artemis.core.server.MessageReference;
+import org.apache.activemq.artemis.core.server.RoutingContext;
+import org.apache.activemq.artemis.core.server.impl.AckReason;
+import org.apache.activemq.artemis.core.server.impl.AddressInfo;
+import org.apache.activemq.artemis.core.server.remotecontrol.MirrorController;
+
+/** this will be used when there are multiple replicas in use. */
+public class AMQPMirrorControllerAggregation implements MirrorController, ActiveMQComponent {
+
+
+   List<AMQPMirrorControllerSource> partitions = new ArrayList<>();
+
+   public void addPartition(AMQPMirrorControllerSource partition) {
+      this.partitions.add(partition);
+   }
+
+   public void removeParition(AMQPMirrorControllerSource partition) {
+      this.partitions.remove(partition);
+   }
+
+
+   @Override
+   public void start() throws Exception {
+
+   }
+
+   @Override
+   public void stop() throws Exception {
+
+   }
+
+   @Override
+   public boolean isStarted() {
+      return false;
+   }
+
+   @Override
+   public void addAddress(AddressInfo addressInfo) throws Exception {
+      for (MirrorController partition : partitions) {
+         partition.addAddress(addressInfo);
+      }
+
+   }
+
+   @Override
+   public void deleteAddress(AddressInfo addressInfo) throws Exception {
+      for (MirrorController partition : partitions) {
+         partition.deleteAddress(addressInfo);
+      }
+   }
+
+   @Override
+   public void createQueue(QueueConfiguration queueConfiguration) throws Exception {
+      for (MirrorController partition : partitions) {
+         partition.createQueue(queueConfiguration);
+      }
+   }
+
+   @Override
+   public void deleteQueue(SimpleString addressName, SimpleString queueName) throws Exception {
+      for (MirrorController partition : partitions) {
+         partition.deleteQueue(addressName, queueName);
+      }
+   }
+
+   @Override
+   public void sendMessage(Message message, RoutingContext context, List<MessageReference> refs) {
+      for (MirrorController partition : partitions) {
+         partition.sendMessage(message, context, refs);
+      }
+   }
+
+   @Override
+   public void postAcknowledge(MessageReference ref, AckReason reason) throws Exception {
+      for (MirrorController partition : partitions) {
+         partition.postAcknowledge(ref, reason);
+      }
+   }
+
+   @Override
+   public void startAddressScan() throws Exception {
+      for (MirrorController partition : partitions) {
+         partition.startAddressScan();
+      }
+   }
+
+   @Override
+   public void endAddressScan() throws Exception {
+      for (MirrorController partition : partitions) {
+         partition.startAddressScan();
```

Comment:
what it calls on partition seems questionable

Reply:
oops.. thanks